### PR TITLE
Volume works

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rio/issues/53

File: cli/pkg/up/up.go [hunk 1]
```diff
@@ -13,23 +17,43 @@ import (
...
 )
 
-func readFileInPath(cwd, file string) ([]byte, error) {
+func readFileInPath(cwd, file, filePath string) ([]byte, error) {
+	if strings.HasPrefix(filePath, "http") {
+		base, err := url.Parse(filePath)
+		if err != nil {
+			return nil, err
+		}
+		ref, err := url.Parse(filepath.Join(file))
```
Comment (ibuildthecloud) [inline cli/pkg/up/up.go]: when dealing with URLs use path.Join

Comment (ibuildthecloud) [reply cli/pkg/up/up.go]: wait, you are joining one element....

File: cli/pkg/up/up.go [hunk 2]
```diff
@@ -13,23 +17,43 @@ import (
 	template2 "github.com/rancher/rio/pkg/template"
 )
 
-func readFileInPath(cwd, file string) ([]byte, error) {
+func readFileInPath(cwd, file, filePath string) ([]byte, error) {
```
Comment (ibuildthecloud) [inline cli/pkg/up/up.go]: Seems like `cwd` should be either a http:// url or local directory.  I don't think you need a third argument.

File: cli/pkg/up/up.go [hunk 3]
```diff
@@ -13,23 +17,43 @@ import (
...
 	f, err := symlink.FollowSymlinkInScope(file, cwd)
 	if err != nil {
 		return nil, err
 	}
 	return ioutil.ReadFile(f)
 }
 
-func readFiles(files []string, template *template2.Template, promptReplaceFile bool) error {
+func readFiles(filePath string, files []string, template *template2.Template, promptReplaceFile bool) error {
 	cwd, err := os.Getwd()
```
Comment (ibuildthecloud) [inline cli/pkg/up/up.go]: remove this getwd code and have the caller pass in the "relativeTo" which is either `http` url or a assumed to be a local directory

File: stacks/istio-stack.yml [hunk 4]
```diff
@@ -138,3 +138,59 @@ services:
         - INSTANCE_IP=$(self/ip)
         - ISTIO_META_POD_NAME=$(self/name)
         secrets: identity:/etc/certs
+
```
Comment (ibuildthecloud) [inline stacks/istio-stack.yml]: Why did this change?

**Pull request closed without merging**